packages/pluggableWidgets/datagrid-date-filter-web/src/components/DatePicker.tsx

```diff
@@ -47,18 +50,26 @@ export const DatePicker = forwardRef(
             }, 10);
         }, []);
 
+        const dateFormats = props.dateFormat ? [props.dateFormat] : undefined;
+        // Check is date format only contains short versions of day and month
+        if (!!props.dateFormat && /(?<![dm])([dm])(?!\1)/.test(props.dateFormat)) {
+            // Replace with full patterns d -> dd, M -> MM
+            dateFormats?.push(props.dateFormat.replaceAll(/(?<!d)d(?!d)|(?<!M)M(?!M)/g, m => m + m));
+        }
+
         return (
             <Fragment>
                 {Portal}
                 <span className="sr-only" id={`${props.id}-label`}>
                     {props.screenReaderInputCaption}
                 </span>
                 <DatePickerComponent
+                    adjustDateOnChange
```

Comment:
in what way does this adjust the format on change?

Reply:
It forces the date to update to the correct format after (1st in the array) after the value changes in the input